azerothcore/azerothcore-wotlk
# fix(Core/Creature): Fix an issue with incorrect corpse decay calculation.

Co-authored by: UltraNix <80540499+UltraNix@users.noreply.github.com>

## Changes Proposed:
- Changes a calculation from removing a corpse immediately after it's looted with certain configs to more correct calculation.
- Changes default corpse decay configs to blizzlike values.

## Issues Addressed:

- Closes #11710
- Closes #14325

## SOURCE:

TBC Classic.

## Tests Performed:

- Tested in-game with `Corpse.Decay.NORMAL` set to 300 and `Rate.Corpse.Decay.Looted` set to 1.

## How to Test the Changes:

1. Change the `Corpse.Decay.*` and `Rate.Corpse.Decay.Looted` config options to whatever values you like.
2. Make sure the creature's corpse is removed at the correct times according to the configs you set.

## Known Issues and TODO List:

- There is an issue with creature respawn times as well, but that's not something I care to deal with right now.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
